pandas/core/groupby/generic.py

```diff
@@ -963,18 +963,23 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
                 # try to treat as if we are passing a list
                 try:
                     result = self._aggregate_multiple_funcs([func], _axis=self.axis)
-                except ValueError as err:
-                    if "no results" not in str(err):
-                        # raised directly by _aggregate_multiple_funcs
-                        raise
-                    result = self._aggregate_frame(func)
-                else:
+
                     # select everything except for the last level, which is the one
                     # containing the name of the function(s), see GH 32040
                     result.columns = result.columns.rename(
                         [self._selected_obj.columns.name] * result.columns.nlevels
                     ).droplevel(-1)
 
+                except ValueError as err:
```

Comment:
if you leave this unchanged, does it work to put another try/except in the else

Reply:
* I was trying to avoid that because nesting exception handling increase complexity.
* But you are right. There is a chance that the `self._aggregate_multiple_funcs` also raise `AttributeError` and this will change the behavior in that case.
* Will update the code as your suggestion. Thanks!

Reply:
I agree that nesting exception handling is not ideal, maybe an isinstance check for a Series instead.

Reply:
between nesting exception handling and explicit type-checking (also nesting inside another exception handling), I think the first one is less evil :). what do you think?

Reply:
from a static typing perspective, the isinstance is probably better. In this case, an AttributeError should be easily avoided. hasattr is not ideal from a static typing/mypy perspective either. However, leave it as a try/except for now and see what others think.